File: PRM392_ClothingStore_MB/app/src/main/java/com/example/prm392_clothingstore_mb/model/ProductDTO.java
package com.example.prm392_clothingstore_mb.model;

import java.io.Serializable;
import java.util.List;

public class ProductDTO implements Serializable {
    private int id;
    private String name;
    private String description;
    private double price;
    private Integer category;
    private String size;
    private String color;
    private String imageUrl;
    private List<CartItemDTO> cartItems;
    private CategoryDTO categoryNavigation;
    private List<OrderItemDTO> orderItems;
    public ProductDTO() {
    }
    public ProductDTO(int id, String name, String description, double price, Integer category, String size, String color, String imageUrl) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.category = category;
        this.size = size;
        this.color = color;
        this.imageUrl = imageUrl;
    }
    public int getId() {
        return id;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }
    public double getPrice() {
        return price;
    }
    public void setPrice(double price) {
        this.price = price;
    }
    public Integer getCategory() {
        return category;
    }
    public void setCategory(Integer category) {
        this.category = category;
    }
    public String getSize() {
        return size;
    }
    public void setSize(String size) {
        this.size = size;
    }
    public String getColor() {
        return color;
    }
    public void setColor(String color) {
        this.color = color;
    }
    public String getImageUrl() {
        return imageUrl;
    }
    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
    public CategoryDTO getCategoryNavigation() {
        return categoryNavigation;
    }
    public void setCategoryNavigation(CategoryDTO categoryNavigation) {
        this.categoryNavigation = categoryNavigation;
    }
}

File: PRM392_ClothingStore_BE/BLL/DTOs/ProductDTO.cs
using System.Collections.Generic;

namespace BLL.DTOs;

public class ProductDTO
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public int? Category { get; set; }
    public string? Size { get; set; }
    public string? Color { get; set; }
    public string? ImageUrl { get; set; }
    public int Stock { get; set; }
    public virtual CategoryDTO? CategoryNavigation { get; set; }
    // Removed circular references to prevent serialization issues
    // public virtual ICollection<CartItemDTO> CartItems { get; set; } = new List<CartItemDTO>();
    // public virtual ICollection<OrderItemDTO> OrderItems { get; set; } = new List<OrderItemDTO>();
}